contrib/build-venv.sh

```diff
@@ -3,7 +3,7 @@
 VENV=$(dirname $0)/..
 BUILD=${VENV}/build
 DIST=${VENV}/dist
-EXTRA_ARGS="-Dsystemd=disabled -Dlaunchd=disabled"
+EXTRA_ARGS="-Dlibxmlb:gtkdoc=false -Dsystemd=disabled -Dlaunchd=disabled"
```

Comment:
the commit message is missing the _why_? i.e. are we going to catch invalid gtk-doc headers in the library sections if we only find out when building the distro packages?

Reply:
With `origin/main`, failing to `build-fwupd` because the errors below:

<details>

<summary>(venv) $ build-fwupd</summary>

```shell
The Meson build system
Version: 1.3.2
Source dir: /home/dm/mywork/fwupd
Build dir: /home/dm/mywork/fwupd/venv/build
Build type: native build
Project name: fwupd
Project version: 2.0.0
C compiler for the host machine: cc (gcc 13.2.0 "cc (Ubuntu 13.2.0-23ubuntu4) 13.2.0")
C linker for the host machine: cc ld.bfd 2.42
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program git found: YES (/usr/bin/git)
Compiler for C supports arguments -Wfatal-errors: YES 
Compiler for C supports arguments -Waggregate-return: YES 
Compiler for C supports arguments -Wunused: YES 
Compiler for C supports arguments -Warray-bounds: YES 
Compiler for C supports arguments -Wcast-align: YES 
Compiler for C supports arguments -Wclobbered: YES 
Compiler for C supports arguments -Wdeclaration-after-statement: YES 
Compiler for C supports arguments -Wdiscarded-qualifiers: YES 
Compiler for C supports arguments -Wduplicated-branches: YES 
Compiler for C supports arguments -Wduplicated-cond: YES 
Compiler for C supports arguments -Wempty-body: YES 
Compiler for C supports arguments -Wfloat-equal: YES 
Compiler for C supports arguments -Wformat=2: YES 
Compiler for C supports arguments -Wformat-nonliteral: YES 
Compiler for C supports arguments -Wformat-security: YES 
Compiler for C supports arguments -Wformat-signedness: YES 
Compiler for C supports arguments -Wignored-qualifiers: YES 
Compiler for C supports arguments -Wimplicit-function-declaration: YES 
Compiler for C supports arguments -Wimplicit-int: YES 
Compiler for C supports arguments -Wincompatible-pointer-types: YES 
Compiler for C supports arguments -Winit-self: YES 
Compiler for C supports arguments -Wint-conversion: YES 
Compiler for C supports arguments -Wlogical-op: YES 
Compiler for C supports arguments -Wmaybe-uninitialized: YES 
Compiler for C supports arguments -Wmissing-declarations: YES 
Compiler for C supports arguments -Wmissing-format-attribute: YES 
Compiler for C supports arguments -Wmissing-include-dirs: YES 
Compiler for C supports arguments -Wmissing-noreturn: YES 
Compiler for C supports arguments -Wmissing-parameter-type: YES 
Compiler for C supports arguments -Wmissing-prototypes: YES 
Compiler for C supports arguments -Wnested-externs: YES 
Compiler for C supports arguments -Wno-cast-function-type: YES 
Compiler for C supports arguments -Wno-address-of-packed-member: YES 
Compiler for C supports arguments -Wno-unknown-pragmas: YES 
Compiler for C supports arguments -Wno-missing-field-initializers: YES 
Compiler for C supports arguments -Wno-strict-aliasing: YES 
Compiler for C supports arguments -Wno-suggest-attribute=format: YES 
Compiler for C supports arguments -Wno-typedef-redefinition: NO 
Compiler for C supports arguments -Wno-unknown-warning-option: NO 
Compiler for C supports arguments -Wno-unused-parameter: YES 
Compiler for C supports arguments -Wno-nonnull-compare: YES 
Compiler for C supports arguments -Wno-analyzer-use-of-uninitialized-value: YES 
Compiler for C supports arguments -Wno-analyzer-fd-double-close: YES 
Compiler for C supports arguments -Wold-style-definition: YES 
Compiler for C supports arguments -Woverride-init: YES 
Compiler for C supports arguments -Wpointer-arith: YES 
Compiler for C supports arguments -Wredundant-decls: YES 
Compiler for C supports arguments -Wreturn-type: YES 
Compiler for C supports arguments -Wshadow: YES 
Compiler for C supports arguments -Wsign-compare: YES 
Compiler for C supports arguments -Wstrict-aliasing: YES 
Compiler for C supports arguments -Wstrict-prototypes: YES 
Compiler for C supports arguments -Wswitch-default: YES 
Compiler for C supports arguments -Wtype-limits: YES 
Compiler for C supports arguments -Wundef: YES 
Compiler for C supports arguments -Wuninitialized: YES 
Compiler for C supports arguments -Wunused-but-set-variable: YES 
Compiler for C supports arguments -Wunused-variable: YES 
Compiler for C supports arguments -Wvla: YES 
Compiler for C supports arguments -Wwrite-strings: YES 
Compiler for C supports link arguments -Wl,-z,relro: YES 
Compiler for C supports link arguments -Wl,-z,defs: YES 
Compiler for C supports link arguments -Wl,-z,now: YES 
Compiler for C supports link arguments -Wl,-z,ibt,-z,shstk: YES 
Program diff found: YES (/usr/bin/diff)
Found pkg-config: YES (/usr/bin/pkg-config) 1.8.1
Run-time dependency gio-2.0 found: YES 2.80.0
Run-time dependency gio-unix-2.0 found: YES 2.80.0
Run-time dependency gmodule-2.0 found: YES 2.80.0
Run-time dependency gudev-1.0 found: YES 238
Dependency xmlb found: NO found 0.3.18 but need: '>= 0.3.19'
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency xmlb found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency xmlb

Executing subproject libxmlb 

libxmlb| Project name: libxmlb
libxmlb| Project version: 0.3.19
libxmlb| C compiler for the host machine: cc (gcc 13.2.0 "cc (Ubuntu 13.2.0-23ubuntu4) 13.2.0")
libxmlb| C linker for the host machine: cc ld.bfd 2.42
libxmlb| Compiler for C supports arguments -Wno-nonnull-compare: YES (cached)
libxmlb| Compiler for C supports arguments -Wno-aggregate-return: YES
libxmlb| Compiler for C supports arguments -Wunused: YES (cached)
libxmlb| Compiler for C supports arguments -Warray-bounds: YES (cached)
libxmlb| Compiler for C supports arguments -Wcast-align: YES (cached)
libxmlb| Compiler for C supports arguments -Wclobbered: YES (cached)
libxmlb| Compiler for C supports arguments -Wdeclaration-after-statement: YES (cached)
libxmlb| Compiler for C supports arguments -Wduplicated-branches: YES (cached)
libxmlb| Compiler for C supports arguments -Wduplicated-cond: YES (cached)
libxmlb| Compiler for C supports arguments -Wempty-body: YES (cached)
libxmlb| Compiler for C supports arguments -Wformat=2: YES (cached)
libxmlb| Compiler for C supports arguments -Wformat-nonliteral: YES (cached)
libxmlb| Compiler for C supports arguments -Wformat-security: YES (cached)
libxmlb| Compiler for C supports arguments -Wformat-signedness: YES (cached)
libxmlb| Compiler for C supports arguments -Wignored-qualifiers: YES (cached)
libxmlb| Compiler for C supports arguments -Wimplicit-function-declaration: YES (cached)
libxmlb| Compiler for C supports arguments -Wincompatible-pointer-types-discards-qualifiers: NO
libxmlb| Compiler for C supports arguments -Winit-self: YES (cached)
libxmlb| Compiler for C supports arguments -Wlogical-op: YES (cached)
libxmlb| Compiler for C supports arguments -Wmissing-declarations: YES (cached)
libxmlb| Compiler for C supports arguments -Wmissing-format-attribute: YES (cached)
libxmlb| Compiler for C supports arguments -Wmissing-include-dirs: YES (cached)
libxmlb| Compiler for C supports arguments -Wmissing-noreturn: YES (cached)
libxmlb| Compiler for C supports arguments -Wmissing-parameter-type: YES (cached)
libxmlb| Compiler for C supports arguments -Wmissing-prototypes: YES (cached)
libxmlb| Compiler for C supports arguments -Wnested-externs: YES (cached)
libxmlb| Compiler for C supports arguments -Wno-cast-function-type: YES (cached)
libxmlb| Compiler for C supports arguments -Wno-error=cpp: YES
libxmlb| Compiler for C supports arguments -Wno-unknown-pragmas: YES (cached)
libxmlb| Compiler for C supports arguments -Wno-discarded-qualifiers: YES
libxmlb| Compiler for C supports arguments -Wno-missing-field-initializers: YES (cached)
libxmlb| Compiler for C supports arguments -Wno-strict-aliasing: YES (cached)
libxmlb| Compiler for C supports arguments -Wno-suggest-attribute=format: YES (cached)
libxmlb| Compiler for C supports arguments -Wno-unused-parameter: YES (cached)
libxmlb| Compiler for C supports arguments -Wnull-dereference: YES
libxmlb| Compiler for C supports arguments -Wold-style-definition: YES (cached)
libxmlb| Compiler for C supports arguments -Woverride-init: YES (cached)
libxmlb| Compiler for C supports arguments -Wpointer-arith: YES (cached)
libxmlb| Compiler for C supports arguments -Wredundant-decls: YES (cached)
libxmlb| Compiler for C supports arguments -Wreturn-type: YES (cached)
libxmlb| Compiler for C supports arguments -Wshadow: YES (cached)
libxmlb| Compiler for C supports arguments -Wsign-compare: YES (cached)
libxmlb| Compiler for C supports arguments -Wstrict-aliasing: YES (cached)
libxmlb| Compiler for C supports arguments -Wstrict-prototypes: YES (cached)
libxmlb| Compiler for C supports arguments -Wswitch-default: YES (cached)
libxmlb| Compiler for C supports arguments -Wtype-limits: YES (cached)
libxmlb| Compiler for C supports arguments -Wundef: YES (cached)
libxmlb| Compiler for C supports arguments -Wuninitialized: YES (cached)
libxmlb| Compiler for C supports arguments -Wunused-but-set-variable: YES (cached)
libxmlb| Compiler for C supports arguments -Wunused-variable: YES (cached)
libxmlb| Compiler for C supports arguments -Wwrite-strings: YES (cached)
libxmlb| Compiler for C supports link arguments -Wl,-z,relro: YES (cached)
libxmlb| Compiler for C supports link arguments -Wl,-z,now: YES (cached)
libxmlb| Dependency gio-2.0 found: YES 2.80.0 (cached)
libxmlb| Dependency gio-unix-2.0 found: YES 2.80.0 (cached)
libxmlb| Run-time dependency liblzma found: YES 5.4.5
libxmlb| Run-time dependency libzstd found: YES 1.5.5
libxmlb| Configuring config.h using configuration
libxmlb| Program python3 found: YES (/home/dm/mywork/fwupd/venv/bin/python3)
libxmlb| Configuring libxmlb.test using configuration
libxmlb| Configuring xb-version.h using configuration
libxmlb| Compiler for C supports link arguments -Wl,--version-script,/home/dm/mywork/fwupd/subprojects/libxmlb/src/libxmlb.map: YES
libxmlb| Configuring xb-tool.1 using configuration
libxmlb| Run-time dependency gobject-introspection-1.0 found: YES 1.80.1
libxmlb| Dependency gobject-introspection-1.0 found: YES 1.80.1 (cached)
libxmlb| Program /usr/bin/x86_64-linux-gnu-g-ir-scanner found: YES (/usr/bin/x86_64-linux-gnu-g-ir-scanner)
libxmlb| Dependency gobject-introspection-1.0 found: YES 1.80.1 (cached)
libxmlb| Program /usr/bin/x86_64-linux-gnu-g-ir-compiler found: YES (/usr/bin/x86_64-linux-gnu-g-ir-compiler)
libxmlb| Program diff found: YES (/usr/bin/diff)
libxmlb| Program gtkdoc-scan found: NO

subprojects/libxmlb/meson.build:184:15: ERROR: Program 'gtkdoc-scan' not found or not executable

A full log can be found at /home/dm/mywork/fwupd/venv/build/meson-logs/meson-log.txt

```

</details>

Reply:
Ohh, I think you want `-Dlibxmlb:gtkdoc=false` instead.

Reply:
Yup, I thought this change similiar to 2399a3d.